pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -282,6 +287,16 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return r.syncDeletedClusterDeployment(cd, hiveImage, cdLog)
 	}
 
+	if cd.Spec.State == hivev1.ClusterDeploymentStateAbsent {
+		cdLog.Info("Absent state detected on clusterdeployment. Deleting...")
+		err := r.Delete(context.TODO(), cd)
+		if err != nil {
+			cdLog.WithError(err).Errorf("cannot delete clusterdeployment")
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{}, nil
+	}
```

Comment:
If I'm reading this right, it looks like if we roll this out while UHC is still just deleting the CD, their code will no longer work and clusters will start piling up.

I think we need an interim hack here where if we see a deletionTimestamp, but state!=absent, then assume to set the state=absent so we get our deprovisioning as before. Then we need to file a card with SDA, wait for the fix to be in stage and prod, then we can kill off the hack and assume everyone is deleting properly.

Reply:
No, that is not the case. You can still delete cluster deployments and they will go away. The only thing that has changed is how we go about deleting them.

Reply:
Ah I see so it's state=absent = assumed cd delete, which is what creates the request.

So I guess eventually there would be a phase 3 where you must set state = absent, and we wait until UHC is ready.

Reply:
Kind of... phase 3 is just removing rbac permission to delete clusterdeployments to every role except the hive manager role.